cloudinit/sources/DataSourceVMware.py

```diff
@@ -527,6 +669,73 @@ def advertise_local_ip_addrs(host_info, rpctool, rpctool_fn):
         LOG.info("advertised local ipv6 address %s in guestinfo", local_ipv6)
 
 
+def advertise_update_events(
+    supported_update_events, enabled_update_events, rpctool, rpctool_fn
+):
+    """
+    advertise_update_events publishes the types of supported and
+    enabled events to guestinfo.
+
+    The string returned from this method adheres to the following
+    format:
+
+        SCOPE=TYPE[;TYPE][,SCOPE=TYPE[;TYPE]]
+
+    For example:
+
+        * network=boot;hotplug
+        * network=boot-new-instance
+
+    The only supported event scope at the moment is network, but there
+    may be more scopes in the future, and this format will support them.
+    """
+    if not rpctool or not rpctool_fn:
+        return None, None
+
+    def get_events_string(events):
+        event_scopes_and_types_list = []
+        for event_scope in events:
+            event_scope_and_types = "{}=".format(event_scope)
+            event_types = []
+            for event_type in events[event_scope]:
+                event_types.append("{}".format(event_type))
+            event_types.sort()
+            event_scope_and_types += ";".join(event_types)
+            event_scopes_and_types_list.append(event_scope_and_types)
+        if len(event_scopes_and_types_list) > 0:
+            event_scopes_and_types_list.sort()
+            return ",".join(event_scopes_and_types_list)
+        return None
+
+    supported_events_string = get_events_string(supported_update_events)
+    if supported_events_string:
+        guestinfo_set_value(
+            SUPPORTED_UPDATE_EVENTS_GUEST_INFO_KEY,
+            supported_events_string,
+            rpctool,
+            rpctool_fn,
+        )
+        LOG.info(
+            "advertised supported update events in guestinfo: %s",
+            supported_events_string,
+        )
+
+    enabled_events_string = get_events_string(enabled_update_events)
+    if enabled_events_string:
+        guestinfo_set_value(
+            ENABLED_UPDATE_EVENTS_GUEST_INFO_KEY,
+            enabled_events_string,
+            rpctool,
+            rpctool_fn,
+        )
+        LOG.info(
+            "advertised enabled update events in guestinfo: %s",
+            enabled_events_string,
+        )
+
+    return supported_events_string, enabled_events_string
```

Comment:
These returns don't get used, right? Can we remove them?

Reply:
They are used in the tests.